luxjs/forms/utils.js

```diff
@@ -3,54 +3,290 @@
+                            shouldScroll = remaining <= height(container) * scrollDistance + 1;
+
+                        if (shouldScroll) {
+                            scope.$apply(function() {
+                                $parse(attrs.selectInfinity)(scope);
+                            });
+                        }
+                    };
+
+                    removeThrottle = throttle('scroll', 'optimizedScroll', container[0]);
+                    container.on('optimizedScroll', handler);
+
+                    scope.$on('$destroy', function() {
+                        removeThrottle();
+                        container.off('optimizedScroll', handler);
+                    });
+
+                    return true;
+                }
+
+                var unbindWatcher = scope.$watch('$select.open', function(newItems) {
+                    if (!newItems) {
+                        return;
+                    }
+
+                    $timeout(function() {
+                        if (tryToSetupInfinityScroll()) {
+                            unbindWatcher();
+                        }
+                    });
+                });
+            }
+        };
+    }])
+    //
+    .directive('remoteOptions', ['$lux', '$timeout', 'remoteService', function ($lux, $timeout, remoteService) {
 
         function fill(api, target, scope, attrs) {
 
-            var id = attrs.remoteOptionsId || 'id',
-                nameOpts = attrs.remoteOptionsValue ? JSON.parse(attrs.remoteOptionsValue) : {
+            var config = {
+                id: attrs.remoteOptionsId || 'id',
+                nameOpts: attrs.remoteOptionsValue ? JSON.parse(attrs.remoteOptionsValue) : {
                     type: 'field',
                     source: 'id'
                 },
-                nameFromFormat = nameOpts.type === 'formatString',
-                initialValue = {},
-                params = JSON.parse(attrs.remoteOptionsParams || '{}'),
-                options = [];
-
-            scope[target.name] = options;
+                initialValue: {},
+                params: JSON.parse(attrs.remoteOptionsParams || '{}'),
+                optionsTotal: 0,
+                queryInitial: {
+                    limit: 25,
+                    offset: 0
+                }
+            };
 
-            initialValue.id = '';
-            initialValue.name = 'Loading...';
+            config.nameFromFormat = config.nameOpts.type === 'formatString';
 
-            options.push(initialValue);
+            remoteService.setupQueryInitial(config);
 
             // Set empty value if field was not filled
             if (scope[scope.formModelName][attrs.name] === undefined)
                 scope[scope.formModelName][attrs.name] = '';
 
-            api.get(null, params).then(function (data) {
-                if (attrs.multiple) {
-                    options.splice(0, 1);
+            remoteService.query(api, target, scope, attrs, config, null, false);
```

Comment:
Here we are calling `remoteService.query` with `null` for the `searchValue` argument, but in the function itself it's comparing `searchValue` with `undefined`.